libraries/AP_InertialSensor/AP_InertialSensor_MPU6000.cpp

```diff
@@ -299,9 +299,22 @@ bool AP_InertialSensor_MPU6000::_init()
 
 void AP_InertialSensor_MPU6000::_fifo_reset()
 {
+    /* We read first the user control register status to
+       prevent a reset to I2C Master if it was enabled.
+       We need to do it here because it's obviously that FIFO was enabled
+       previously.
+    */
+    uint8_t user_ctrl = _register_read(MPUREG_USER_CTRL);
+
+    /* Reset the FIFO buffer */
     _register_write(MPUREG_USER_CTRL, 0);
     _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_FIFO_RESET);
     _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_FIFO_EN);
+
+    /* We come back from last user control register status and
+       continue the normal hardware operation.
+    */
+    _register_write(MPUREG_USER_CTRL, user_ctrl);
```

Comment:
Yep, you are right, that bitwise procedure work perfectly because isolate to the others control bits in the register and optimize the code.

Reworking and fixing it...